fetch.bs

```diff
@@ -1876,18 +1876,45 @@ with given <var>strategy</var>, <var>pull</var> action and <var>cancel</var> act
 are optional, run these steps:
 
 <ol>
- <li><p>Let <var>init</var> be a new object.
+ <li><p>Let <var>startAlg</var> be an algorithm that returns <code>undefined</code>.
 
- <li><p>Set <var>init</var>["pull"] to a function that runs <var>pull</var> if <var>pull</var> is
- given.
+ <li><p>Let <var>pullAlg</var> be an algorithm that returns:
 
- <li><p>Set <var>init</var>["cancel"] to a function that runs <var>cancel</var> if
- <var>cancel</var> is given.
+ <dl class=switch>
+ <dt>If <var>pull</var> is given
+ <dd>the result of [=promise-calling=] <var>pull</var>().
```

Comment:
@annevk usually objects to using the promises guide. However, I don't see much alternative except for copying the phrasing that the promises guide uses.

Reply:
Very out-of-context question: What is the reasoning for objection to using the promises guide? Haven't read all of it, just curious (sorry this is not the best place).

Reply:
@domfarolino Possibly https://github.com/w3ctag/promises-guide/issues/52.

I know of one trap in there: because [aggregating promises](https://www.w3.org/2001/tag/doc/promises-guide#aggregating-promises) uses `Promise.all()` it calls the current (_not_ original) value of `Promise.reject()` which may have been replaced. This makes it unsafe to use in algorithms that are not re-entrant or are otherwise vulnerable to unexpected JS execution.There may be other traps I don't know about.

Having said that, we use it in the streams standard and I think it is fine if you use it with care.